services/server/config/config.go

```diff
@@ -239,12 +249,14 @@ func LoadConfig(path string, out *Config) error {
 func loadConfigFile(path string) (*Config, error) {
 	config := &Config{}
 
-	file, err := toml.LoadFile(path)
+	f, err := os.Open(path)
 	if err != nil {
-		return nil, fmt.Errorf("failed to load TOML: %s: %w", path, err)
+		return nil, err
 	}
+	defer f.Close()
 
-	if err := file.Unmarshal(config); err != nil {
+	// TODO: Add DisallowUnknownFields, requires better testing and bubbling errors
+	if err := toml.NewDecoder(f).Decode(config); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal TOML: %w", err)
```

Comment:
Just in case you don't already know, you can type-assert the error returned by `Decode` into a [`toml.DecodeError`](https://pkg.go.dev/github.com/pelletier/go-toml/v2#DecodeError), which allows you to print a more human friendly error ([example](https://pkg.go.dev/github.com/pelletier/go-toml/v2#example-DecodeError)).

Reply:
Thanks, added better error handling using the toml error types, thanks.